mp4parse/src/lib.rs

```diff
@@ -1398,7 +1471,8 @@ fn read_audio_sample_entry<T: Read>(src: &mut BMFFBox<T>, track: &mut Track) ->
                 }
 
                 let esds = try!(read_esds(&mut b));
-                codec_specific = Some(esds);
+                track.codec_type = esds.audio_codec;
```

Comment:
For consistency, set track.codec_type and codec_specific in the same order in each of the branches - right now there's a mix of both variants.